hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/ConsensusValidatorImpl.java

```diff
@@ -74,56 +74,70 @@ public void validate(Collection<FileStreamSignature> signatures) throws Signatur
-                    continue;
-                }
-
-                nodeAccountIdToStakeMap.put(nodeAccountId, nodeStake.getStake());
+        var nodeIdToNodeAccountIdMap = addressBook.getNodeIdNodeAccountIdMap();
+        for (var nodeStake : nodeStakes) {
+            var nodeAccountId = nodeIdToNodeAccountIdMap.get(nodeStake.getNodeId());
+            if (nodeAccountId == null) {
+                throw new SignatureVerificationException(String.format("Invalid Node Id %s. Consensus not " +
+                        "reached for file %s", nodeStake.getNodeId(), filename));
+            }
+
+            long stake = nodeStake.getStake();
+            totalStake += stake;
+            nodeAccountIdToStakeMap.put(nodeAccountId, stake);
+        }
+
+        if (totalStake == 0) {
+            // Count the node stakes as signatures
+            nodeAccountIdToStakeMap.clear();
+            // Use the number of nodes as the total stake.
+            totalStake = nodeIdToNodeAccountIdMap.size();
+
+            if (totalStake == 0) {
+                throw new SignatureVerificationException(String.format("Invalid total staking weight. Consensus not " +
+                        "reached for file %s", filename));
             }
         }
 
+        var stakeRequiredForConsensus = getStakeRequiredForConsensus(totalStake);
+        long debugStake = 0;
         long consensusCount = 0;
         for (String key : signatureHashMap.keySet()) {
             var validatedSignatures = signatureHashMap.get(key);
             long stake = 0L;
             for (var signature : validatedSignatures) {
-                // If the map has no entry for the node account id, a default value of 1 is used to count a signature.
+                // If the stake cannot be found in the map, count it as a signature giving it a value of 1.
                 stake += nodeAccountIdToStakeMap.getOrDefault(signature.getNodeAccountId(), 1L);
             }
 
-            if (canReachConsensus(stake, totalStake)) {
+            if (canReachConsensus(stake, stakeRequiredForConsensus)) {
                 consensusCount += validatedSignatures.size();
                 validatedSignatures.forEach(s -> s.setStatus(FileStreamSignature.SignatureStatus.CONSENSUS_REACHED));
             }
+
+            if (debugStake < stake) {
+                debugStake = stake;
+            }
         }
 
         if (consensusCount > 0) {
             return;
         }
 
+        log.debug("Highest encountered Stake: {}, Total Stake: {}", debugStake, totalStake);
+        log.debug("Consensus Ratio: {}", commonDownloaderProperties.getConsensusRatio());
+        log.debug("Stake Required For Consensus: {}", stakeRequiredForConsensus);
+        log.debug("Result: {}", canReachConsensus(debugStake, stakeRequiredForConsensus));
         throw new SignatureVerificationException(String.format("Consensus not reached for file %s", filename));
```

Comment:
4 debug logs is too much and should be consolidated. But instead I think we should remove all the logs and add just the critical info to the exception message. The other info can be derived from application properties and computations from just stake and total stake.

```suggestion
        throw new SignatureVerificationException(String.format("Consensus not reached for file %s with %d/%d stake", filename));
```